src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs

```diff
@@ -74,16 +73,22 @@ public JsonWebToken(string jwtEncodedString)
         }
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="JsonWebToken"/> class where the header contains the crypto algorithms applied to the encoded header and payload. The jwtEncodedString is the result of those operations.
+        /// Initializes a new instance of the <see cref="JsonWebToken"/> class where the header contains the crypto algorithms applied to the encoded header and payload.
         /// </summary>
-        /// <param name="header">Contains JSON objects representing the cryptographic operations applied to the JWT and optionally any additional properties of the JWT.</param>
-        /// <param name="payload">Contains JSON objects representing the claims contained in the JWT. Each claim is a JSON object of the form { Name, Value }.</param>
+        /// <param name="header">A string containing JSON which represents the cryptographic operations applied to the JWT and optionally any additional properties of the JWT.</param>
+        /// <param name="payload">A string containing JSON which represents the claims contained in the JWT. Each claim is a JSON object of the form { Name, Value }.</param>
         /// <exception cref="ArgumentNullException">'header' is null.</exception>
         /// <exception cref="ArgumentNullException">'payload' is null.</exception>
-        public JsonWebToken(JObject header, JObject payload)
+        public JsonWebToken(string header, string payload)
         {
-            Header = header ?? throw LogHelper.LogArgumentNullException(nameof(header));
-            Payload = payload ?? throw LogHelper.LogArgumentNullException(nameof(payload));
+            if (string.IsNullOrEmpty(header))
+                throw LogHelper.LogArgumentNullException(nameof(header));
+
+            if (string.IsNullOrEmpty(payload))
+                throw LogHelper.LogArgumentNullException(nameof(payload));
+
+            Header = JObject.Parse(header);
```

Comment:
Would it make sense to capture an exception here and log & re-throw _our_ exception, in case that provided json string is invalid?  _JObject.Parse_ throws _JsonReaderException_ when an error occurs during parsing.
Same applies for _Payload_.

Reply:
@GeoK I agree, that would be cleaner.
We throw ArgumentException IDX12723 and IDX12724 when parsing fails in JwtSecurityToken.
Are you asking for a new exception OR stick with ArgumentException?

Reply:
Good suggestion. I decided to throw an ArgumentException just to be consistent with what JwtSecurityToken does.